lancero_source.go

```diff
@@ -282,26 +291,58 @@ func (ls *LanceroSource) Sample() error {
 	ls.mixRequests = make(chan *MixFractionObject, MIXDEPTH)
 	ls.currentMix = make(chan []float64, MIXDEPTH)
 
-	ls.rowColCodes = make([]RowColCode, ls.nchan)
-	i := 0
-	for _, device := range ls.active {
-		cardNchan := device.ncols * device.nrows * 2
-		for j := 0; j < cardNchan; j += 2 {
-			col := j / (2 * device.nrows)
-			row := (j % (2 * device.nrows)) / 2
-			ls.rowColCodes[i+j] = rcCode(row, col, device.nrows, device.ncols)
-			ls.rowColCodes[i+j+1] = ls.rowColCodes[i+j]
+	// Check that ls.chanSepColumns and chanSepCards are appropriate,
+	// i.e. large enough to avoid channel number collisions.
+	if ls.chanSepColumns > 0 {
+		for _, device := range ls.active {
+			if device.nrows > ls.chanSepColumns {
+				log.Printf("Warning: /dev/lancero%d has %d rows, which exceeds ChanSepColumns (%d). Setting latter to 0.",
+							device.devnum, device.nrows, ls.chanSepColumns)
+				ls.chanSepColumns = 0
```

Comment:
This is silent error handling, return an error instead.

Reply:
Also it seems like we want an assert is 0 or positive?

Reply:
Errors now returned here and in the next comment. Also added an error if either `chanSepCards` or Columns is negative.